File: Pemrograman Berorientasi Objek/KUIS2 PBO Kelas/SOAL 2/Tabel.cpp
class Atrribut;

class Tabel{
	
	private:
		int baris;
		int kolom;
	
	public:
		/*konstruktor*/
		Tabel(){
		}
		
		Tabel(int b,int k){
			this->baris = b;
			this->kolom = k;
		}
		
		/*method get dan set*/
		void setBaris(int b){
			this->baris = b;
		}
		
		int getBaris(){
			return this->baris;
		}
		
		void setKolom(int k){
			this->kolom = k;
		}
		
		int getKolom(){
			return this->kolom;
		}
		
		/*untuk membuat tabel*/
		void buatBaris(string isi[],int add,int n){
			
			for(int i=0;i<n;i++){
				for(int j=0;j<isi[i].length();j++){
					cout << "-" << ends;
				}
			}
			cout << endl;
			
			for(int i=0;i<n;i++){
				cout << "| " << ends;
				cout << isi[i] << ends;
				for(int j=0;j<(add-3);j++){
					cout << " " << ends;
				}
				cout << "|" << ends;
			}
			cout << endl;
		}
		
		/*mengakses di kelas atribut*/
		friend class Atrribut;
	
};

File: Pemrograman Berorientasi Objek/Materi/Pertemuan 3/Latihan/java/Tabel.java
public class Tabel{
	
	private int baris;
	//banyaknya baris
	
	private int kolom;
	//banyaknya kolom
	
	Tabel(){
	}
	
	Tabel(int baris, int kolom){
		//konstruktor langsung mengisi atribut
		this.baris = baris;
		this.kolom = kolom;
	}
	
	public void setBaris(int baris){
		//mengeset
		this.baris = baris;
	}
	
	public void buatHeader(String[] isi,int add){
		int i = 0, j = 0;//inisialisasi variabel di java
		
		//buat tabel bagian atas
		for(i=0;i<isi.length;i++){
			for(j=0;j<(isi[i].length()+add);j++){
				System.out.print("-");
			}
		}
		System.out.println("");
		
		//buat baris isi
		for(i=0;i<isi.length;i++){
			System.out.print("| ");
			System.out.print(isi[i]);
			for(j=0;j<(add-3);j++){
				System.out.print(" ");
			}
			System.out.print("|");
		}
		System.out.println("");		
	}
	
	public void buatBaris(String[] isi,int add){
		int i = 0, j = 0;//inisialisasi variabel di java
		
		//buat tabel bagian atas
		for(i=0;i<isi.length;i++){
			for(j=0;j<(isi[i].length()+add);j++){
				System.out.print("-");
			}
		}
		System.out.println("");
		
		//buat baris isi
		for(i=0;i<isi.length;i++){
			System.out.print("| ");
			System.out.print(isi[i]);
			for(j=0;j<(add-3);j++){
				System.out.print(" ");
			}
			System.out.print("|");
		}
		System.out.println("");
		
	}
}